core/src/verified_vote_packets.rs

```diff
@@ -158,12 +225,49 @@ impl VerifiedVotePackets {
                     let slot = vote.last_voted_slot().unwrap();
                     let hash = vote.hash();
 
-                    let validator_votes = self.0.entry(vote_account_key).or_default();
-                    validator_votes.insert((slot, hash), (packet_batch, signature));
-
-                    if validator_votes.len() > MAX_VOTES_PER_VALIDATOR {
-                        let smallest_key = validator_votes.keys().next().cloned().unwrap();
-                        validator_votes.remove(&smallest_key).unwrap();
+                    match (vote, is_full_tower_vote_enabled) {
+                        (VoteStateUpdate(_), true) => {
+                            let latest_slot_sent_to_bank = match self.0.get(&vote_account_key) {
```

Comment:
nit: this isn't the latest slot sent to the bank, but the latest slot detected in gossip right? Maybe rename to `latest_gossip_vote`?

Same nit applies to the function name `get_latest_slot_sent_to_bank()`